### DISCUSSION TITLE
can I pass a under-estimate SDF to Manifold::LevelSet?

### DISCUSSION THREAD

**Discussion by wtholliday** [elalish/manifold#977] [Q&A] [answered]
I've got an approximate SDF which has the same zero level set, but under-estimates distances away from the surface. Will `Manifold::LevelSet` still generate a good mesh (albeit a bit slower) or does it need an exact SDF?

Asking because I'd like to get rid of my own Marching Cubes implementation which produces some degenerate triangles in output.

**Comment by elalish** [comment, marked as answer]
Yes, in fact our current levelset doesn't assume a distance function at all - the sign is all that's needed. We've been talking about optimizing it for the case where the function is known to be a distance or an underestimate, but haven't done so yet.

**Reply by wtholliday** [reply]
Oh that's interesting... so does it sample SDF values on a uniform grid?

**Reply by pca006132** [reply]
@wtholliday yes
